pymc/distributions/multivariate.py

```diff
@@ -2167,3 +2174,183 @@ def logp(value, mu, W, alpha, tau):
             tau > 0,
             msg="-1 <= alpha <= 1, tau > 0",
         )
+
+
+class StickBreakingWeightsRV(RandomVariable):
+    name = "stick_breaking_weights"
+    ndim_supp = 1
+    ndims_params = [0, 0]
+    dtype = "floatX"
+    _print_name = ("StickBreakingWeights", "\\operatorname{StickBreakingWeights}")
+
+    def make_node(self, rng, size, dtype, alpha, K):
+
+        alpha = at.as_tensor_variable(alpha)
+        K = at.as_tensor_variable(intX(K))
+
+        if K.eval() < 0:
+            raise ValueError("K needs to be positive.")
```

Comment:
If I write `if at.lt(K, 0)`, it often seems to incorrectly evaluate to True... Is there a better way to write this `if` statement?

Reply:
Here you can't raise a ValueError yet because you the values are possibly still symbolic.  Such error should be in the `rng_fn` where it is already concrete.

In contrast you can raise erros for ndims, because those are fixed.